### ISSUE TITLE
Version 2.0.0 build fail

### ISSUE THREAD

**Issue by Pro-pra** [dbus-cxx/dbus-cxx#73]
`[ 22%] Building CXX object CMakeFiles/dbus-cxx.dir/dbus-cxx/messageiterator.cpp.o
/usr/bin/g++ -Ddbus_cxx_EXPORTS -Dinterface=struct -I/builddir/build/BUILD/dbus-cxx-2.0.0 -I/builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx -I/builddir/build/BUILD/dbus-cxx-2.0.0/build -I/builddir/build/BUILD/dbus-cxx-2.0.0/build/dbus-cxx -I/usr/include/sigc++-3.0 -I/usr/lib64/sigc++-3.0/include -O2 -fomit-frame-pointer -gdwarf-4 -Wstrict-aliasing=2 -pipe -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -fstack-protector-strong --param=ssp-buffer-size=4 -m64 -mtune=generic -std=gnu++11 -fno-operator-names -fPIC -pthread -std=gnu++17 -MD -MT CMakeFiles/dbus-cxx.dir/dbus-cxx/messageiterator.cpp.o -MF CMakeFiles/dbus-cxx.dir/dbus-cxx/messageiterator.cpp.o.d -o CMakeFiles/dbus-cxx.dir/dbus-cxx/messageiterator.cpp.o -c /builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx/messageiterator.cpp
In file included from /builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx/messageiterator.h:20,
                 from /builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx/messageiterator.cpp:19:
/builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx/demangle.h: In function 'std::string DBus::demangle()':
/builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx/demangle.h:48:39: error: must '#include <typeinfo>' before using 'typeid'
   24 | #include <dbus-cxx/dbus-cxx-config.h>
  +++ |+#include <typeinfo>
   25 | 
......
   48 |     std::string arg1_name = typeid( T ).name();
      |                                       ^`

**Comment by Pro-pra** [comment]
error: must '#include <typeinfo>' before using 'typeid'

**Comment by Pro-pra** [comment]
and after fix another fail

```
[  7%] Building CXX object CMakeFiles/dbus-cxx.dir/dbus-cxx/connection.cpp.o
/usr/bin/g++ -Ddbus_cxx_EXPORTS -Dinterface=struct -I/builddir/build/BUILD/dbus-cxx-2.0.0 -I/builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx -I/builddir/build/BUILD/dbus-cxx-2.0.0/build -I/builddir/build/BUILD/dbus-cxx-2.0.0/build/dbus-cxx -I/usr/include/sigc++-3.0 -I/usr/lib64/sigc++-3.0/include -O2 -fomit-frame-pointer -gdwarf-4 -Wstrict-aliasing=2 -pipe -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fPIC -fstack-protector-strong --param=ssp-buffer-size=4 -m64 -mtune=generic -std=gnu++11 -fno-operator-names -fPIC -pthread -std=gnu++17 -MD -MT CMakeFiles/dbus-cxx.dir/dbus-cxx/connection.cpp.o -MF CMakeFiles/dbus-cxx.dir/dbus-cxx/connection.cpp.o.d -o CMakeFiles/dbus-cxx.dir/dbus-cxx/connection.cpp.o -c /builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx/connection.cpp
/builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx/connection.cpp: In member function 'DBus::DispatchStatus DBus::Connection::dispatch()':
/builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx/connection.cpp:566:36: error: no match for 'operator!=' (operand types are 'std::thread::id' and 'std::thread::id')
  566 |     if( std::this_thread::get_id() != m_priv->m_dispatchingThread ) {
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                 |             |
      |                                 |             std::thread::id
      |                                 std::thread::id
In file included from /usr/include/c++/11.1.0/iosfwd:40,
                 from /usr/include/c++/11.1.0/ios:38,
                 from /usr/include/c++/11.1.0/istream:38,
                 from /usr/include/c++/11.1.0/sstream:38,
                 from /builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx/signal.h:19,
                 from /builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx/connection.h:22,
                 from /builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx/connection.cpp:19:
/usr/include/c++/11.1.0/bits/postypes.h:227:5: note: candidate: 'template<class _StateT> bool std::operator!=(const std::fpos<_StateT>&, const std::fpos<_StateT>&)'
  227 |     operator!=(const fpos<_StateT>& __lhs, const fpos<_StateT>& __rhs)
      |     ^~~~~~~~
```

**Comment by Pro-pra** [comment]
error: no match for 'operator!='

**Comment by rm5248** [comment]
I tested with GCC-11 and I don't get the first error(typeinfo).  I did find a few compilation errors and fixed those.  Do you have a patch for the first error?

**Comment by Pro-pra** [comment]
Thank you! i use gcc 10.3 and typeinfo need for me.

Also with patches build fail on link stage:
`DEBUG: /usr/bin/x86_64-openmandriva-linux-gnu-ld: CMakeFiles/dbus-cxx-glib.dir/glibdispatcher.cpp.o: in function `DBus::GLib::GLibDispatcher::~GLibDispatcher()':
DEBUG: /builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx-glib/glibdispatcher.cpp:45: undefined reference to `g_io_channel_unref'
DEBUG: /usr/bin/x86_64-openmandriva-linux-gnu-ld: CMakeFiles/dbus-cxx-glib.dir/glibdispatcher.cpp.o: in function `DBus::GLib::GLibDispatcher::create_connection(DBus::BusType)':
DEBUG: /builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx-glib/glibdispatcher.cpp:54: undefined reference to `DBus::Connection::create(DBus::BusType)'
DEBUG: /usr/bin/x86_64-openmandriva-linux-gnu-ld: /builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx-glib/glibdispatcher.cpp:55: undefined reference to `DBus::Connection::bus_register()'
DEBUG: /usr/bin/x86_64-openmandriva-linux-gnu-ld: CMakeFiles/dbus-cxx-glib.dir/glibdispatcher.cpp.o: in function `DBus::GLib::GLibDispatcher::create_connection(std::string)':
DEBUG: /builddir/build/BUILD/dbus-cxx-2.0.0/dbus-cxx-glib/glibdispatcher.cpp:62: undefined reference to `DBus::Connection::create(std::string)'
`
disable -no-undefined help me

**Comment by rm5248** [comment]
What distribution are you using?  I can't reproduce on Debian using either version 10.2 or 11.1.

**Comment by Pro-pra** [comment]
x86_64-openmandriva-linux-gnu-ld

**Comment by krab** [comment]
```
/home/krab/Downloads/dbus-cxx-git/unit-tests/tools-tests/signal-proxy.cpp: In function ‘int main(int, char**)’:
/home/krab/Downloads/dbus-cxx-git/unit-tests/tools-tests/signal-proxy.cpp:42:27: error: ‘sleep_for’ is not a member of ‘std::this_thread’
   42 |         std::this_thread::sleep_for( std::chrono::seconds( 1 ) );
      |                           ^~~~~~~~~
```
Same issue on pop os which based on Ubuntu 22.04 LTS. GCC 11.2.0.

**Comment by Pro-pra** [comment]
fixed in 2.1.0 release
